packages/desktop-client/src/components/transactions/table/utils.ts

```diff
@@ -0,0 +1,138 @@
+import { parseISO, isValid as isDateValid } from 'date-fns';
+
+import { evalArithmetic } from 'loot-core/shared/arithmetic';
+import { currentDay } from 'loot-core/shared/months';
+import {
+  amountToInteger,
+  type CurrencyAmount,
+  integerToCurrency,
+} from 'loot-core/shared/util';
+import {
+  type AccountEntity,
+  type CategoryEntity,
+  type TransactionEntity,
+} from 'loot-core/types/models';
+
+export type SerializedTransaction = Omit<TransactionEntity, 'date'> & {
+  date: string;
+  debit: CurrencyAmount;
+  credit: CurrencyAmount;
+};
+
+export type TransactionEditFunction = (
+  id: TransactionEntity['id'],
+  name: string,
+) => void;
+
+export type TransactionUpdateFunction = <T extends keyof SerializedTransaction>(
+  name: T,
+  value: SerializedTransaction[T],
+) => void;
+
+export function serializeTransaction(
+  transaction: TransactionEntity,
+  showZeroInDeposit?: boolean,
+): SerializedTransaction {
+  const { amount, date: originalDate } = transaction;
+
+  let debit = amount < 0 ? -amount : null;
+  let credit = amount > 0 ? amount : null;
+
+  if (amount === 0) {
+    if (showZeroInDeposit) {
+      credit = 0;
+    } else {
+      debit = 0;
+    }
+  }
+
+  let date = originalDate;
+  // Validate the date format
+  if (!isDateValid(parseISO(date))) {
+    // Be a little forgiving if the date isn't valid. This at least
+    // stops the UI from crashing, but this is a serious problem with
+    // the data. This allows the user to go through and see empty
+    // dates and manually fix them.
+    console.error(`Date ‘${date}’ is not valid.`);
+    date = null as unknown as string;
```

Comment:
A little worried about this crashing other places down the line. Is it very annoying to allow `null` in `SerializedTransaction` for `date`? If so we should at least add a TODO to fix it later IMO